core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -59,7 +63,7 @@
 /**
  * Unit test for {@link org.apache.calcite.rel.externalize.RelJson}.
  */
-public class RelWriterTest {
+public class RelWriterTest extends SqlToRelTestBase {
   public static final String XX = "{\n"
```

Comment:
I know maybe you need some abstractions in SqlToRelTestBase, but do we really need to make this class extend SqlToRelTestBase ? This is writer test and seems has no relationship with sql-to-rel conversion.

Reply:
Make this class extend SqlToRelTestBase to reuse the created Tester instance. The constructor of TesterImpl is protected, cannot just create an instance of TesterImpl without extending SqlToRelTestBase.

Reply:
All you need is just a relNode, you can construct it with a `RelBuilder` or new the RelNode directly.

Reply:
I think this will make things complicated, it's better to choose a simple and  straight forward way to get the relnode tree.

Reply:
There is already some code snippet to construct the `RelNode` directly [1], so what do you mean by "things complicated" ?

[1] https://github.com/apache/calcite/blob/18974ce8a945424a8a3465c03b253e292641f420/core/src/test/java/org/apache/calcite/plan/RelWriterTest.java#L313

Reply:
I know there exists sample code snippet to do this. But what I mean is, we already have a simpler way to get the RelNode, why take the effort to build it manually.

Reply:
Can you take a look at what is `SqlToRelTestBase` used for ? Not just say that "Oh, this extend works for me!".